### ISSUE TITLE
Add piecewise regression example

### ISSUE THREAD

**Issue by tomicapretto** [bambinos/bambi#437]
We can define 

```python
def truncate(x, l):
    x = (x - l) * (x >= l)
    return x
```

And then simply do

```python
formula = "response ~ x + truncate(x, 10) + truncate(age, 20)"
```

**Comment by hadjipantelis** [comment]
Being somewhat thick here, wouldn't be a smoother (no pun intended) thing to use `bs` and set the `degree` and `knots` accordingly? That way, the user could have a linear spline to create a piecewise regression.

**Comment by hadjipantelis** [comment]
Being somewhat thick here, wouldn't be a smoother (no pun intended) option to use `bs` and set the `degree` and `knots` accordingly? That way, the user could have a linear spline to create a piecewise regression.

**Comment by tomicapretto** [comment]
Yes, you could also do that, but the following points come to my mind:

* You need to know or understand what a spline basis is.
* In many cases (or most?) piecewise linear regression is not introduced as a special case of a basis spline.
* This example would also show that you can write and use your own functions in the model formula.

However, your point is also a motivation to show the connection between piecewise regression using `truncate()` and a `bs()` using linear functions. That could be a good example.

**Comment by hadjipantelis** [comment]
Fair points. I appreciate that my proposition might be a tad bullish regarding people's familiarity with splines use.

**Comment by tomicapretto** [comment]
I like when people get involved in the issues and propose new ideas. I really appreciate your comments. So please feel free to share when you want to propose something.

To be a little clearer, I'm not saying that we should only have the example using `truncated()`. What I'm saying is that we could have both, and explain the connection between them 😉

**Comment by ivanmkc** [comment]
@tomicapretto I tried using a custom function but I get this error:

```
[KeyError: 'truncate']()
```

**Comment by tomicapretto** [comment]
@ivanmkc could you share an example of what you're trying to do? What you're suggesting as the problem seems to make sense though.
